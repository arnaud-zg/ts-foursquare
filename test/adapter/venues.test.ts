import {
  adaptGetVenuesSearch,
  adaptGetVenuesExplore,
} from '../../src/adapter/venues'
import { payload as payloadGetVenuesSearch } from '../epics/__mocks__/getVenuesSearchAsync.resolve'
import { payload as payloadGetVenuesExplore } from '../epics/__mocks__/getVenuesExploreAsync.resolve'

describe('adapter/venues/adaptGetVenuesSearch', () => {
  it('should get an empty list', () => {
    adaptGetVenuesSearch({
      ...payloadGetVenuesSearch,
      meta: {
        code: 400,
        requestId: payloadGetVenuesSearch.meta.requestId,
      },
    })
  })

  it('should get a list of venues', () => {
    expect(adaptGetVenuesSearch(payloadGetVenuesSearch).length).toEqual(
      payloadGetVenuesSearch.response.venues.length
    )
  })
})

describe('adapter/venues/adaptGetVenuesExplore', () => {
  it('should return null', () => {
    expect(
      adaptGetVenuesExplore({
        ...payloadGetVenuesExplore,
        meta: {
          code: 400,
          requestId: payloadGetVenuesSearch.meta.requestId,
        },
      })
    ).toEqual([])
  })

  it('should get result', () => {
    expect(adaptGetVenuesExplore(payloadGetVenuesExplore)).toMatchSnapshot()
  })
})
