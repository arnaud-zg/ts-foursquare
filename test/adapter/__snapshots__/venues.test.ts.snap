// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adapter/venues/adaptGetVenuesExplore should get result 1`] = `
Array [
  Object {
    "items": Array [
      Object {
        "reasons": Object {
          "count": 0,
          "items": Array [
            Object {
              "reasonName": "globalInteractionReason",
              "summary": "This spot is popular",
              "type": "general",
            },
          ],
        },
        "venue": Object {
          "categories": Array [
            Object {
              "icon": Object {
                "prefix": "https://ss3.4sqi.net/img/categories_v2/shops/food_gourmet_",
                "suffix": ".png",
              },
              "id": "4bf58dd8d48988d1f5941735",
              "name": "Gourmet Shop",
              "pluralName": "Gourmet Shops",
              "primary": true,
              "shortName": "Gourmet",
            },
          ],
          "id": "49b6e8d2f964a52016531fe3",
          "location": Object {
            "address": "179 E Houston St",
            "cc": "US",
            "city": "New York",
            "country": "United States",
            "crossStreet": "btwn Allen & Orchard St",
            "distance": 130,
            "formattedAddress": Array [
              "179 E Houston St (btwn Allen & Orchard St)",
              "New York, NY 10002",
              "United States",
            ],
            "labeledLatLngs": Array [
              Object {
                "label": "display",
                "lat": 40.72286707707289,
                "lng": -73.98829148466851,
              },
            ],
            "lat": 40.72286707707289,
            "lng": -73.98829148466851,
            "postalCode": "10002",
            "state": "NY",
          },
          "name": "Russ & Daughters",
          "venuePage": Object {
            "id": "77298563",
          },
        },
      },
    ],
    "name": "recommended",
    "type": "Recommended Places",
  },
]
`;
